File: Algorithms/Java/Integer to Roman.java
public class Solution {
    public String intToRoman(int num) {
        int[] integer = { 1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1 };
		String[] roman = { "M", "CM", "D", "CD", "C", "XC", "L", "XL", "X",
				"IX", "V", "IV", "I" };
		String res = "";
		int i = 0;
		int amount;
		while (true) {
			amount = num / integer[i];
			num %= integer[i];
			while (amount > 0) {
				res += roman[i];
				amount--;
			}
			i++;
			if (num == 0) {
				break;
			}
		}
		return res;
        
    }
}

File: Algorithms/Python/Integer-to-Roman.py
class Solution(object):

    def intToRoman(self, num):
        """
        :type num: int
        :rtype: str
        """
        dic = {1: 'I', 4: 'IV', 5: 'V', 9: 'IX', 10: 'X',
               40: 'XL', 50: 'L', 90: "XC", 100: 'C', 400: 'CD', 500: 'D', 900: 'CM', 1000: 'M'}
        ans = ''
        s = dic.keys()
        s.sort(reverse=True)
        for x in s:
            while num >= x:
                ans += dic[x]
                num -= x
        return ans
